extension/encoding/awslogsencodingextension/internal/unmarshaler/cloudtraillogs/unmarshaler.go

```diff
@@ -0,0 +1,230 @@
+	}
+
+	return u.processRecords(cloudTrailLogs.Records)
+}
+
+func (u *CloudTrailLogsUnmarshaler) processRecords(records []CloudTrailRecord) (plog.Logs, error) {
+	logs := plog.NewLogs()
+
+	if len(records) == 0 {
+		return logs, nil
+	}
+
+	// Create a single resource logs entry for all records
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	scopeLogs.Scope().SetName(metadata.ScopeName)
+	scopeLogs.Scope().SetVersion(u.buildInfo.Version)
+
+	// Set resource attributes based on the first record
+	// (all records have the same account ID and region)
+	u.setResourceAttributes(resourceLogs.Resource().Attributes(), records[0])
+
+	for _, record := range records {
+		logRecord := scopeLogs.LogRecords().AppendEmpty()
+		if err := u.setLogRecord(logRecord, record); err != nil {
+			return plog.Logs{}, err
+		}
+	}
+
+	return logs, nil
+}
+
+func (u *CloudTrailLogsUnmarshaler) setResourceAttributes(attrs pcommon.Map, record CloudTrailRecord) {
+	attrs.PutStr(string(conventions.CloudProviderKey), conventions.CloudProviderAWS.Value.AsString())
+	attrs.PutStr(string(conventions.CloudRegionKey), record.AwsRegion)
+	attrs.PutStr(string(conventions.CloudAccountIDKey), record.RecipientAccountID)
+}
+
+func (u *CloudTrailLogsUnmarshaler) setLogRecord(logRecord plog.LogRecord, record CloudTrailRecord) error {
+	t, err := time.Parse(time.RFC3339, record.EventTime)
+	if err != nil {
+		return fmt.Errorf("failed to parse timestamp of log: %w", err)
+	}
+	logRecord.SetTimestamp(pcommon.NewTimestampFromTime(t))
+	u.setLogAttributes(logRecord.Attributes(), record)
+	return nil
+}
+
+func (u *CloudTrailLogsUnmarshaler) setLogAttributes(attrs pcommon.Map, record CloudTrailRecord) {
+	attrs.PutStr("aws.cloudtrail.event_version", record.EventVersion)
+
+	attrs.PutStr("aws.cloudtrail.event_id", record.EventID)
+
+	if record.EventName != "" {
+		attrs.PutStr(string(conventions.RPCMethodKey), record.EventName)
+	}
+
+	attrs.PutStr(string(conventions.RPCSystemKey), record.EventType)
+
+	if record.EventSource != "" {
+		attrs.PutStr(string(conventions.RPCServiceKey), record.EventSource)
+	}
+
+	if record.RequestID != "" {
+		attrs.PutStr(string(conventions.AWSRequestIDKey), record.RequestID)
+	}
+
+	attrs.PutStr("aws.event.category", record.EventCategory)
+
+	if record.ReadOnly != nil {
+		attrs.PutBool("aws.event.read_only", *record.ReadOnly)
+	}
+
+	if record.ManagementEvent != nil {
+		attrs.PutBool("aws.event.management", *record.ManagementEvent)
+	}
+
+	if record.SourceIPAddress != "" {
+		attrs.PutStr(string(conventions.SourceAddressKey), record.SourceIPAddress)
+	}
+
+	if record.UserAgent != "" {
+		attrs.PutStr(string(conventions.UserAgentOriginalKey), record.UserAgent)
+	}
+
+	if record.SessionCredentialFromConsole == "true" {
+		attrs.PutBool("aws.session.console", true)
+	}
+
+	if record.UserIdentity != nil {
+		if principalID, ok := record.UserIdentity["principalId"].(string); ok {
```

Comment:
Any security concern about adding those identity fields to the logs attributes by default? It seems cloudtrail redacts the `userName` already when it's necessary, but I'm not sure about the other ones.

Reply:
https://aws.amazon.com/blogs/security/modifications-to-aws-cloudtrail-event-data-of-iam-identity-center/#:~:text=Effective%20July%2014%2C%202025%2C%20IAM,provides%20guidance%20for%20these%20updates.

Reply:
My 2c: I don't think it's the role of the encoding extension to redact things. If a user wants to redact, they can use the filter or transform processors for example.

Reply:
Yes I agree, I didn't mean redacting the values, just avoiding adding them to the log attributes. Anyway, If you folks think they are not a security concern, I guess we can keep it as it's and just add the extra IAM fields.